src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -128,6 +131,8 @@ citus_add_local_table_to_metadata_internal(Oid relationId, bool cascadeViaForeig
 								  "to 'off' to disable this behavior")));
 	}
 
+	ErrorIfPartitionTableAddedToMetadata(relationId);
```

Comment:
This function errors out if the given relation is a partition. I think we can't move it to `ErrorIfUnsupportedCreateCitusLocalTable` as it would error out when converting a partition. I think it should stay here at `citus_add_local_table_to_metadata_internal`, so that we can error out only if the user tries to create citus local tables with the partition, instead of the parent.

Reply:
alright, makes sense